lib/psu_identity/underscore.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class String
+  def underscore
+    gsub(/::/, '/')
+      .gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
+      .gsub(/([a-z\d])([A-Z])/, '\1_\2')
+      .tr('-', '_')
+      .downcase
+  end
```

Comment:
Is this just a copy of https://api.rubyonrails.org/classes/String.html#method-i-underscore ? Might be nice to just make a comment stating that is the case if so. Clever solution, BTW!

Reply:
I yanked this from (https://stackoverflow.com/Questions/1509915/converting-camel-case-to-underscore-case-in-ruby) which, at one point is what ActiveSupport was using, I think it's since diverged to (https://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-underscore) Which is a good question, when loaded into a rails app, would we get `underscore` for free? Do we ever intend on using outside of a rails app? How do other gems handle this kind of thing? @rschenk ??

Reply:
We wouldn't be able to test the parts of the gem that require `rails`, so I don't think it would be a good idea to treat this like it will always be loading into a rails app.